adaptors/dummy/test/e2e/usability/usability_test.go

```diff
@@ -0,0 +1,552 @@
+	var deviceLinkKey = types.NamespacedName{
+		Namespace: deviceLink.Namespace,
+		Name:      deviceLink.Name,
+	}
+	if err := k8sCli.Get(testCtx, deviceLinkKey, &deviceLink); err != nil {
+		GinkgoT().Log(err)
+		return
+	}
+	patch := []byte(`{"spec":{"template":{"spec":{"gear":"wrong-gear"}}}}`)
+	Expect(k8sCli.Patch(testCtx, &deviceLink, client.RawPatch(types.MergePatchType, patch))).Should(Succeed())
+}
+
+func isDeviceConnectedTrue() {
+	var targetKey = types.NamespacedName{
+		Namespace: deviceLink.Namespace,
+		Name:      deviceLink.Name,
+	}
+	Eventually(func() bool {
+		if err := k8sCli.Get(testCtx, targetKey, &deviceLink); err != nil {
+			GinkgoT().Log(err)
+			return false
+		}
+		return deviceLink.GetDeviceConnectedStatus() == metav1.ConditionTrue
+	}, 300, 1).Should(BeTrue())
+}
+
+func isDeviceConnectedFalse() {
+	var targetKey = types.NamespacedName{
+		Namespace: deviceLink.Namespace,
+		Name:      deviceLink.Name,
+	}
+	Eventually(func() bool {
+		if err := k8sCli.Get(testCtx, targetKey, &deviceLink); err != nil {
+			GinkgoT().Log(err)
+			return false
+		}
+		return deviceLink.GetDeviceConnectedStatus() == metav1.ConditionFalse
+	}, 300, 1).Should(BeTrue())
+}
+
+func editGearToFast() {
+	var deviceLinkKey = types.NamespacedName{
+		Namespace: deviceLink.Namespace,
+		Name:      deviceLink.Name,
+	}
+	if err := k8sCli.Get(testCtx, deviceLinkKey, &deviceLink); err != nil {
+		GinkgoT().Log(err)
+		return
+	}
+	patch := []byte(`{"spec":{"template":{"spec":{"gear":"fast"}}}}`)
+	Expect(k8sCli.Patch(testCtx, &deviceLink, client.RawPatch(types.MergePatchType, patch))).Should(Succeed())
+}
+
+func isDeviceConfigFast() {
+	var deviceList apiv1alpha1.DummySpecialDeviceList
+	Eventually(func() bool {
+		if err := k8sCli.List(testCtx, &deviceList, client.InNamespace("default")); err != nil {
+			GinkgoT().Log(err)
+			return false
+		}
+		for _, device := range deviceList.Items {
+			return device.Spec.Gear == "fast"
+		}
+		return false
+	}, 300, 1).Should(BeTrue())
+}
+
+func deleteDummyAdaptorPods() {
+	Expect(k8sCli.DeleteAllOf(testCtx, &corev1.Pod{}, client.InNamespace("octopus-system"), client.MatchingLabels{"app.kubernetes.io/name": "octopus-adaptor-dummy"})).
+		Should(Succeed())
+}
+
+func deleteLimbsPods() {
+	Expect(k8sCli.DeleteAllOf(testCtx, &corev1.Pod{}, client.InNamespace("octopus-system"), client.MatchingLabels{"app.kubernetes.io/component": "limb"})).
+		Should(Succeed())
+}
+
+func isDummyAdaptorPodsError() {
+	length := len("octopus-adaptor-dummy-adaptor")
+	var podList corev1.PodList
+	Eventually(func() bool {
+		if err := k8sCli.List(testCtx, &podList, client.InNamespace("octopus-system"), client.MatchingLabels{"app.kubernetes.io/name": "octopus-adaptor-dummy"}); err != nil {
+			GinkgoT().Log()
+			return false
+		}
+		for _, pod := range podList.Items {
+			if len(pod.Name) > length && pod.Name[:length] == "octopus-adaptor-dummy-adaptor" {
+				return pod.Status.Reason == "Error"
+			}
+		}
```

Comment:
Why we need this condition? Label selector is not enough?

Reply:
updated